tests/sles4sap/netweaver_ascs_install.pm

```diff
@@ -52,8 +52,7 @@ sub run {
 
     # If running in DESKTOP=gnome, systemd-logind restart may cause the graphical console to
     # reset and appear in SUD, so need to select 'root-console' again
-    wait_still_screen;
-    select_console 'root-console';
+    select_console 'root-console' unless (check_screen 'root-console');
```

Comment:
Are you sure this saves time? By default the timeout of `check_screen` is 30 seconds and not really the same behaviour as wait_still_screen. It is more likely for a screen to show up as still than exactly matching one tag so I fear you are wasting 30s here. Also, I do not recommend to use `check_screen` with default timeout anywhere. A better alternative is either `check_screen …, 0` which I think won't work here as you tell that systemd-logind switches away the console *or* a multi-tag assert_screen with `match_has_tag` which might work here

Reply:
I am sure it saves time when `DESKTOP=textmode`. Not so sure in `gnome`. It also reads better semantically.

Checking the logs from the test runs I have:

With `wait_still_screen`:
- Textmode: 12 seconds
- Gnome: 19 seconds

With `check_screen`:
- Textmode: 9 seconds
- Gnome: 40 seconds

Will try the multi-tag `assert_screen` approach. Thanks.